scripts/publish-npm-packages.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+# Set default values for dist-tag and registry for local development
+# to prevent running this script and accidentally publishing to npm
+dist_tag="dev"
+registry="http://localhost:4873"
+
+# shellcheck source=./scripts/helpers/exit-if-fail.sh
+source "$(dirname "$0")/helpers/exit-if-fail.sh"
+
+# check if there were any changes to packages between current and previous commit
+count=$(git diff HEAD~1..HEAD --name-only -- packages | awk '{c++} END {print c}')
+
+# Parse command line arguments
+while [[ $# -gt 0 ]]; do
+    key="$1"
+    case $key in
+        --dist-tag)
+        dist_tag="$2"
+        shift # past argument
+        shift # past value
+        ;;
+        --registry)
+        registry="$2"
+        shift # past argument
+        shift # past value
+        ;;
+        *)    # unknown option
+        echo "Unknown option: $1"
+        exit 1
+        ;;
+    esac
+done
+
+if [ -z "$NPM_TOKEN" ]; then
+  echo "The NPM_TOKEN environment variable does not exist."
+  exit 1
+fi
+
+if [ -z "$count" ]; then
+  echo "No changes in packages, skipping packages publishing"
+else
+  echo "Changes detected in ${count} packages"
+  echo "Starting to release latest canary version"
+
+  echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" >> ~/.npmrc
+
+  echo $'\nPacking packages'
+  yarn packages:pack
+
```

Comment:
if this script is the whole publishing process then I think validation step goes in here after the `yarn packages:pack`

Reply:
The issue with running it here is this script only runs on a merge to main. I think it would be good to run these validations after a successful build of the packages to catch anything with dependency bumps prior to them getting merged.